src/main/java/wooteco/subway/station/dao/StationDao.java

```diff
@@ -36,18 +43,70 @@ public Station insert(Station station) {
         return new Station(id, station.getName());
     }
 
-    public List<Station> findAll() {
-        String sql = "select * from STATION";
-        return jdbcTemplate.query(sql, rowMapper);
+    public List<StationLineResponse> findAll() {
+        String sql = "select S.id AS station_id, S.name AS station_name, L.id AS line_id, L.name AS line_name, L.color AS line_color " +
+                "FROM STATION AS S " +
+                "LEFT JOIN SECTION AS SE ON (SE.up_station_id = S.id OR SE.down_station_id = S.id) " +
+                "LEFT JOIN LINE AS L ON L.id = SE.line_id";
+
+        List<Map<String, Object>> result = jdbcTemplate.queryForList(sql);
+        Map<Long, List<Map<String, Object>>> resultByStation = result.stream()
+                .collect(Collectors.groupingBy(it -> (Long) it.get("station_id")));
+
+        return resultByStation.entrySet()
+                .stream()
+                .map(it -> mapStationLineResponse(it.getValue()))
+                .collect(Collectors.toList());
+    }
+
+    private StationLineResponse mapStationLineResponse(final List<Map<String, Object>> result) {
+        if (result.size() == 0) {
+            throw new RuntimeException();
+        }
+
+        List<TransferLineResponse> transferLineResponses = extractTransferLine(result);
+
+        return new StationLineResponse(
+                (Long) result.get(0).get("station_id"),
+                (String) result.get(0).get("station_name"),
+                transferLineResponses
+        );
+    }
+
+    private List<TransferLineResponse> extractTransferLine(List<Map<String, Object>> result) {
+        if (result.isEmpty() || result.get(0).get("line_id") == null) {
+            return Collections.EMPTY_LIST;
```

Comment:
```suggestion
            return Collections.emptyList();
```
큰 차이는 없지만 요 메서드를 사용하는 제네릭을 반환하니 참고해주세요ㅎㅎ